### PR TITLE
Remove backup EA4 profile if run with --check

### LINKED ISSUE #126: elevate-cpanel --check is not cleaning up the EA4 profile it creates
The `elevate-cpanel` script presently generates a backup of the current EasyApache 4 profile during the evaluation of blockers. It is appropriate to retain this profile when evaluating prior to a real upgrade attempt, but it is not appropriate when only blockers are being evaluated via the `--check` flag. In combination with [the change to cPanel version 106 which runs the script in check mode as part of the Security Advisor](https://docs.cpanel.net/changelogs/106-change-log/#106-0-1), this is spamming the list of EA4 profiles with useless entries.

### PR DESCRIPTION
**Pull request by sloanebernstein**
Prevent unwanted retention of the created file when the script has no intention of performing the upgrade.

Fixes #126 and CPANEL-41603.

By submitting pull requests to this repo, I agree to the Contributor License Agreement which can be found at: https://github.com/cpanel/elevate/blob/main/docs/cPanel-CLA.pdf

### CODE REVIEW

File: elevate-cpanel [hunk 1]
```diff
@@ -2820,7 +2820,9 @@ sub _blocker_ea4_profile ($self) {
 
     INFO("Checking EasyApache profile compatibility with AlmaLinux 8.");
 
-    return unless backup_ea4_profile();
+    my $backup = backup_ea4_profile();
+    return         unless $backup;
+    unlink $backup unless $self->{_abort_on_first_blocker};
```
Comment (atoomic) [inline elevate-cpanel:2825]: this is correct but this logic should belong to `backup_ea4_profile` itself
and we should convert as a method and call it `$self->backup_ea4_profile`

File: elevate-cpanel [hunk 2]
```diff
@@ -1456,7 +1456,7 @@ sub backup_ea4_profile () {
 
     update_stage_file( { ea4 => $data } );
 
-    return 1;
+    return $json;
```
Comment (atoomic) [inline elevate-cpanel:1459]: note: the variable name `$json` is confusing, this is not part of these  changes
but could good to rename it as `ea_json_profile_file` or similar so we know this is a file

**Review by atoomic — CHANGES_REQUESTED** [review]

File: elevate-cpanel (hunk 1 above)
Comment (sloanebernstein) [reply elevate-cpanel]: `backup_ea4_profile` is now a method.

File: elevate-cpanel (hunk 2 above)
Comment (sloanebernstein) [reply elevate-cpanel]: Changed to `$json_path`, if that's any better.

**Comment by atoomic** [comment]
note: conflict due to multiple functions which were converted to methods to add support to Rocky

File: elevate-cpanel [hunk 3]
```diff
@@ -1558,18 +1558,21 @@ sub backup_ea4_profile ($self) {
...
+    my $profile = eval { Cpanel::JSON::LoadFile($json_path) } // {};
     if ( ref $profile->{os_upgrade} && ref $profile->{os_upgrade}->{dropped_pkgs} ) {
         $data->{dropped_pkgs} = $profile->{os_upgrade}->{dropped_pkgs};
     }
 
     update_stage_file( { ea4 => $data } );
 
+    # CPANEL-41603: delete ea4 backup file if in --check mode
+    unlink $json_path unless $self->{_abort_on_first_blocker};
+
```
Comment (atoomic) [inline elevate-cpanel:1575]: do we need to store the profile path in data when `_abort_on_first_blocker`?

File: t/ea4.t [hunk 4]
```diff
@@ -322,6 +326,31 @@ Please remove these packages before continuing the update.
...
+sub test_blocker_file_behavior : Tests(8) ($self) {
+
+    my $ea_info_check = sub {
+        message_seen( 'INFO' => "Checking EasyApache profile compatibility with AlmaLinux 8." );
+    };
+
+    $self->{mock_cpev}->redefine( _get_ea4_profile => sub { $self->{mock_profile}->contents('{}'); return PROFILE_FILE } );
+
+    my $cpev = bless {}, 'cpev';
+    $cpev->{_abort_on_first_blocker} = 1;    # enforce a die
```
Comment (atoomic) [inline t/ea4.t:338]: note this can also be replaced by: `my $cpev = cpev(_abort_on_first_blocker => 1 )`

**Review by atoomic — APPROVED** [review]

File: t/ea4.t (hunk 4 above)
Comment (atoomic) [reply t/ea4.t:338]: applied via e43b899ec861d1d000e3779bebf37075ccd4f82b

**Pull request merged**